### ISSUE TITLE
PyPI's XML-RPC is deprecated

### ISSUE THREAD

**Issue by webknjaz** [sphinx-contrib/spelling#227]
Got this in CI half an hour ago:
```python-traceback
Exception occurred:
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/xmlrpc/client.py", line 668, in close
    raise Fault(**self._stack[0])
xmlrpc.client.Fault: <Fault -32500: 'RuntimeError: PyPI no longer supports the XMLRPC list_packages method. Use JSON or Simple API instead. See https://github.com/pypi/warehouse/issues/16642 and https://warehouse.pypa.io/api-reference/xml-rpc.html#deprecated-methods for more information.'>
The full traceback has been saved in /tmp/sphinx-err-q6f1kt1m.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <[https://github.com/sphinx-doc/sphinx/issues](https://github.com/sphinx-doc/sphinx/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen)>. Thanks!
```
*(https://github.com/ansible/awx-plugins/actions/runs/10774764535/job/29877727274#step:18:45)*

https://warehouse.pypa.io/api-reference/xml-rpc.html#deprecated-methods has
> Use the [Simple API](https://warehouse.pypa.io/api-reference/legacy.html) to query for list of project names with releases on PyPI.

**Comment by webknjaz** [comment]
A CI job rerun succeeded. So they've probably configured hard failures for a certain percentage of requests. FYI.

**Comment by webknjaz** [comment]
It doesn't seem like it can be replaced with JSON API. Perhaps, it's possible to grab this data from the BigQuery dataset instead: https://warehouse.pypa.io/api-reference/bigquery-datasets.html.

Meanwhile, I suppose the current workaround is `spelling_ignore_pypi_package_names = False`.

**Comment by webknjaz** [comment]
The endpoint is now gone for good: https://mail.python.org/archives/list/pypi-announce@python.org/message/5YFJHGOTOUBJJ5ZSFKGN3ABVN4Y7AJ4K/

**Comment by dhellmann** [comment]
It sounds like that feature should be disabled by default until it's clear what alternatives might exist. I'm not likely to have time to do anything about that for a while, but if someone was to submit a PR I could review it.

**Comment by dhellmann** [comment]
This fix is included in release 8.0.1.
